rfc/rfc-62/rfc-62.md

```diff
@@ -0,0 +1,443 @@
+
+Users can use this Diagnostic Reporter Tool in the following two ways：
+1. Users can directly enable this diagnostic reporter in the writing jobs, at this time diagnostic reporter tool will go through current hudi table and generate report files as the last stage after commit.
+2. Users can directly generate the corresponding report file for a hudi table through the hudi cli command.
+
+Report json file location: hudi will create a new meta folder name `.hoodie/report/` for this report result, for example  `.hoodie/report/xxxx.json` and `.hoodie/report/xxxx.zip`
+
+Naming rules for specific report json/zip file: `HudiTableName + "_" + HudiVersion + "_" + appName + "_" + applicationId + "_" + applicationAttemptId + "_" + isLocal + format`.
+There may be multiple writers and writing multiple hudi tables. Using this naming rule should be sufficient to identify a writer.
+
+
+AS for the json file this diagnostic reporter tool generated, it will contains three parts: data information, meta information and engine runtime information(take spark as an example here):
+
+### Data information
+
+Diagnostic reporter tool need to list all the partitions and all the data files for current hudi table. Then do basic analysis, build follow result (part1 of report.json)
+Also we can get these information from metadata table directly if users enable it. And we can run hoodieMetadata table valuation tool after get all these information.
+```json
+{
+    "total":{
+        "partition_numbers":"100",
+        "data_file_numbers":"1000",
+        "data_volume":"1000GB",
+        "partition_keys":"year/month/day"
+    },
+    "max_data_volume_per_partition":{
+        "partition_name":"2022/09/05",
+        "data_file_number":"10",
+        "file_group_number":"5",
+        "data_volume":"10GB",
+        "max_data_file_size":"1GB",
+        "min_data_file_size":"100MB"
+    },
+    "max_data_file_numbers_per_partition":{
+        "partition_name":"2022/09/06",
+        "data_file_number":"100",
+        "file_group_number":"50",
+        "data_volume":"1GB",
+        "max_data_file_size":"100MB",
+        "min_data_file_size":"10MB"
+    },
+    "sample_hoodie_key":{
+        "hoodie_keys":"user_id",
+        "sample_values":[
+            "123",
+            "456"
+        ]
+    }
+}
+```
+
+We can quickly catch up the data distribution characteristics of the current hudi table through this part, which can be used to determine whether there is a small file problem 
+or a data hotspot problem, etc.
+In addition, through data distribution and sample hudi keys, we can also use this information to help users choose the most appropriate index mode, for example:
+- If the hudi data is roughly evenly distributed, users can try using bucket index to improve upsert performance.
+- If the hudi key is monotonically increasing then users can try the bloom index
+
+Of course, the behavior of the sample hoodie key is false by default for privacy reasons
```

Comment:
So we only record the record key characteristics instead of sampling the keys which need to be masked？
We can get the characteristics information from schema directly maybe.